rts/at/integration/src/main/java/org/jboss/narayana/rest/integration/RecoveryManager.java

```diff
@@ -81,6 +86,8 @@ public void persistParticipantInformation(final ParticipantInformation participa
             final OutputObjectState state = getParticipantInformationOutputState(participantInformation);
             final Uid uid = new Uid(participantInformation.getId());
 
+            // to identify the uid from the participant persisted into the object store in order to delete it later
+            committedParticipants.put(participantInformation.getId(), uid);
```

Comment:
I note this is going in the map before the store is updated

Reply:
This was intentional originally as I was thinking that it's safer to try to try a remove an non-existing record from the object store than does not try it at all. I was thinking on some issue being thrown after the record was persisted but the call was not yet returned to this point.
Then the `persistedParticipants` would not be aware.
But when I'm rethinking now it seems to me it's safe to move it after the persistent call.